### PR TITLE
End to end tests for Sign In

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
Adds end to end Cypress tests for the Sign In page

## Why is `returnUrl` todo for social sign in?
Because this is yet to be implemented. I've added a test for this ready for when this feature is working, until then the test is skipped using `skip`.

### CODE REVIEW

File: cypress/integration/ete/sign_in.spec.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  inbox: '21497957-085c-4c84-93eb-e025d1dd0145',
+};
+
+describe('Sign in flow', () => {
+  it('links to the correct places', () => {
+    cy.visit('/signin');
+    cy.contains('Terms and Conditions').should(
+      'have.attr',
+      'href',
+      'https://www.theguardian.com/help/terms-of-service',
```
[the remark this answers is not in the document]
Comment (oliverlloyd) [reply cypress/integration/ete/sign_in.spec.js:15]: Ha, you're totally right, great feedback! I'll make this change.

Note. The other links where we have similar checks link out to external pages so we're limited to checking the href there but in this case we own the page we're linking to so your suggested method makes more sense.

File: cypress/integration/ete/sign_in.spec.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+  it('navigates to registration', () => {
+    cy.visit('/signin');
+    cy.contains('Register').click();
+    cy.get('[data-cy="register-button"]').should('be.visible');
+  });
+
+  it('respects the returnUrl query param', () => {
+    cy.visit(
+      `/signin?returnUrl=${encodeURIComponent(
```
[the remark this answers is not in the document]
Comment (oliverlloyd) [reply cypress/integration/ete/sign_in.spec.js:62]: I think you're right here. The content in this article does change a lot and whilst I would hope the iconic text of Mystery Bird will remain forever 🕵️ 🐦 ❤️  we could also pick something equally iconic which is less likely to be edited.

How about https://www.theguardian.com/world/2013/jun/09/edward-snowden-nsa-whistleblower-surveillance?

File: cypress/integration/ete/sign_in.spec.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    cy.get('input[name=password]').type('password');
+    cy.get('[data-cy="sign-in-button"]').click();
+    cy.contains('This email and password combination is not valid');
+  });
+
+  it('correctly signs in an existing user', () => {
+    cy.visit('/signin');
+    cy.get('input[name=email]').type(existing.email);
+    cy.get('input[name=password]').type(existing.password);
+    cy.get('[data-cy="sign-in-button"]').click();
```
[the remark this answers is not in the document]
Comment (oliverlloyd) [reply cypress/integration/ete/sign_in.spec.js:38]: There's some [great docs](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements) that discuss this. In short, it depends.

File: cypress/integration/ete/sign_in.spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ob6160) [reply cypress/integration/ete/sign_in.spec.js:62]: You're going to have to fill me in on Mystery Bird 😅

**Pull request merged**